### PR TITLE
Implement test filtering by name

### PR DESCRIPTION
**Pull request by lampajr**
## Fixes Issue

Fixes https://github.com/Hyperfoil/Horreum/issues/1412

## Changes proposed

The main purpose of this change was to add a name filtering on AllTests page.
As part of this new feature, I had chance to optimize, clean and refactor all tables trying to remove code duplication as much as possible.

Client-side:
- [x] Rename `Table` to `CustomTable` to differentiate with the Patternfly one
- [x] Use patternfly 5 tables, rather than raw tables
- [x] Reduce number of internal calls to the backend when changing sorting (it was calling the same endpoint two times, thus refreshing the UI two times in a row)
- [x] Performing server-side ordering on AllTests page (in prod it is performing client-side ordering, which is actually a page-isolated ordering)
- [x] Fix Runs ordering by Datasets size (there was wrong accessor)
- [x] Removed patternfly react-deprecated Dropdowns for all table columns, e.g., `watching` and `actions`
- [x] Moved pagination component inside the `CustomTable` itself to avoid duplication

Server-side:
- [x] Added new query parameter named `name` on TestSummary endpoint
- [x] Added test on that method

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by lampajr** [comment]
Below you can find some of the changes, but I'd encourage you to give it a try locally as the changes impacted all places where a Table was/is used:

Filter by name:
image

Improved a little bit the dataset comparison by wrapping it into a PageSection:
image

**Comment by lampajr** [comment]
The only doubt I have is whether we want to add the filtering on the Schemas as well (this would be appreciated, see https://github.com/Hyperfoil/Horreum/issues/1198). I know that all changes regarding the schemas have been put on hold, but I am not expecting huge effort to extend this mechanism to the Schema entities (especially considering that we don't know when we can start the schema refactoring). wdyt @johnaohara ?

File: horreum-web/src/components/CustomTable.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    count?: number,
+    perPage?: number,
+    page?: number,
+    onSetPage?: OnSetPage
+    onPerPageSelect?: OnPerPageSelect,
+    variant?: 'top' | 'bottom' | PaginationVariant,
+    isCompact?: boolean
+}
+
+type CustomTableProps<D extends object> = {
```
Comment (johnaohara) [inline horreum-web/src/components/CustomTable.tsx:38]: Here we define a custom table that is using generics, but the only place we specify a type is in `horreum-web/src/domain/schemas/SchemaList.tsx`

If we are going to use generics to enforce type checking at compile time, I think we should extend the usage to cover all instances of `<CustomTable/>`

**Review by johnaohara — CHANGES_REQUESTED** [review]
@lampajr the only major issue is the Changes modal popup

**Comment by johnaohara** [comment]
> The only doubt I have is whether we want to add the filtering on the Schemas as well (this would be appreciated, see #1198). I know that all changes regarding the schemas have been put on hold, but I am not expecting huge effort to extend this mechanism to the Schema entities (especially considering that we don't know when we can start the schema refactoring). wdyt @johnaohara ?

Lets implement Schema name filtering in a separate PR

**Comment by johnaohara** [comment]
@lampajr please can you squash commits?

**Comment by lampajr** [comment]
> @lampajr please can you squash commits?

Squashed and applied all your suggestions!

File: horreum-web/src/components/CustomTable.tsx (hunk 1 above)
Comment (lampajr) [reply horreum-web/src/components/CustomTable.tsx:38]: Yeah make sense, cleaner this way! 
Added on all usages :heavy_check_mark:

File: horreum-web/src/domain/alerting/ChangeTable.tsx [hunk 2]
```diff
@@ -72,13 +84,13 @@ const ChangeMenu = ({ change, onDelete, onUpdate }: ChangeMenuProps) => {
...
-                onClose={() => setModalChange(undefined)}
-                onUpdate={onUpdate}
-            />
+                <ChangeModal
+                    change={modalChange}
+                    isOpen={!!modalChange}
+                    onClose={() => setModalChange(undefined)}
+                    onUpdate={onUpdate}
+                />
+            </Dropdown>
```
[the remark this answers is not in the document]
Comment (lampajr) [reply horreum-web/src/domain/alerting/ChangeTable.tsx]: I think it was my fault, I wrongly put the modal inside the dropdwon. Fixed now.

**Review by johnaohara — APPROVED** [review]
LGTM

**Pull request merged**